### PR TITLE
Quote Search fixes: lp1784141

### PR DESCRIPTION
**Pull request by daschuer**
Now we can search a quotes string like:

title:"My Love"
It finds "My Love", "My Love (Air Mix)" and "My Lover is Gone";

title:"My Love " <- tailing space
It finds only "My Love (Air Mix)"

This works also for all columns without "title:"

### CODE REVIEW

**Review by uklotzde — CHANGES_REQUESTED** [review]
There are some edge cases that need to be considered (and tested).

File: src/library/searchqueryparser.cpp [hunk 1]
```diff
@@ -123,15 +123,16 @@ void SearchQueryParser::parseTokens(QStringList tokens,
         } else if (m_textFilterMatcher.indexIn(token) != -1) {
             QString field = m_textFilterMatcher.cap(1);
             QString argument = getTextArgument(
-                m_textFilterMatcher.cap(2), &tokens).trimmed();
+                    m_textFilterMatcher.cap(2), &tokens);
 
             if (!argument.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (daschuer) [reply src/library/searchqueryparser.cpp]: I am not sure if it really makes sense, but now you are able to search for titles with a blank.
tiltle:" " matches all titles with a blank in the middle like "My Love".
This is at least now wrong, so there is no need to omit this by special case code.

**Comment by daschuer** [comment]
Done. Thank you for review.

File: src/library/searchqueryparser.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (uklotzde) [reply src/library/searchqueryparser.cpp]: Searching for fields containing multiple subsequent whitespaces definitely makes sense to identify and fix redundant whitespaces. I will try and check this.

**Comment by uklotzde** [comment]
Nice :+1: I was able to identify and fix redundant whitespace in 8 title with the query `title:"  "`

**Comment by uklotzde** [comment]
...but the following _false positives_ are found when searching for titles with double spaces:
`Slow Burnin' 22.5˚ Fahrenheit`
`Komm´ mit`

**Comment by daschuer** [comment]
Ok, we actually have an issue here. Currently you cannot search for "˚" because it is considered as its  base character " ".  I will provide a fix.

**Comment by daschuer** [comment]
Strange "˚" != "°" you test `Slow Burnin' 22.5˚ Fahrenheit` string contains the first, but the second is on the German keyboard.

File: src/library/searchquery.cpp [hunk 2]
```diff
@@ -157,8 +169,16 @@ bool TextFilterNode::match(const TrackPointer& pTrack) const {
 
 QString TextFilterNode::toSql() const {
     FieldEscaper escaper(m_database);
-    QString escapedArgument = escaper.escapeString(kSqlLikeMatchAll + m_argument + kSqlLikeMatchAll);
-
+    QString argument = m_argument;
+    if (argument.size() > 0) {
+        if (argument[argument.size() - 1].isSpace()) {
+            // LIKE eats a tailing space. This can be avoided by adding a '_'
```
Comment (uklotzde) [inline src/library/searchquery.cpp]: "_t **r** ailing_" is the correct term as far as I remember. This typo also appears in the name of a test.

Comment (daschuer) [reply src/library/searchquery.cpp]: .. and I thought is should be "tailing" ... fixed.

**Review by uklotzde — APPROVED** [review]

**Comment by uklotzde** [comment]
Searching now works as expected, even when using composed characters or decoration characters :+1: 

LGTM.

**Comment by uklotzde** [comment]
Merged into 2.2 and resolved a small merge conflict: Replaced `token` with `argument` in `SearchQueryParser::parseTokens()`

**Pull request merged**